Source/Core/Core/PowerPC/JitCommon/Jit_Util.cpp

```diff
@@ -626,7 +605,8 @@ void EmuCodeBlock::WriteToConstRamAddress(int accessSize, OpArg arg, u32 address
 	if (arg.IsImm())
 	{
 		arg = SwapImmediate(accessSize, arg);
-		MOV(accessSize, MDisp(RMEM, address & 0x3FFFFFFF), arg);
+		MOV(32, R(RSCRATCH), Imm32(address));
+		MOV(accessSize, MRegSum(RMEM, RSCRATCH), arg);
```

Comment:
Yes, exactly.

I've considered pointing RMEM at logical_base+0x8000000; in addition to making offsets fit, it allows a neat solution to any issues with RMEM+reg+imm overflow.  It would probably be a codesize increase overall, though.